Origin/bugfix/registration from all pages

Kirill1504 opened the pull request.

Kirill-Lebedev committed:
Добавил модальное окно регистрации на productPage.html, stockGlobalPage.html, stocksPage.html и subcategoryPage.html . Вынес логику регистрации в файл registration.js

Kirill-Lebedev committed:
Добавил модальное окно регистрации на productPage.html, stockGlobalPage.html, stocksPage.html, newsPage.html и subcategoryPage.html . Вынес логику регистрации в файл registration.js

Kirill-Lebedev committed:
Добавил модальное окно регистрации на productPage.html, stockGlobalPage.html, stocksPage.html и subcategoryPage.html . Вынес логику регистрации в файл registration.js

Kirill-Lebedev committed:
Merge remote-tracking branch 'origin/origin/bugfix/registrationFromAllPages' into bugfix/registrationFromAllPages

# Conflicts:
#	src/main/resources/templates/productPage.html

Kirill-Lebedev committed:
Merge branch 'dev' into bugfix/registrationFromAllPages

# Conflicts:
#	src/main/resources/templates/stocksPage.html

Kirill-Lebedev committed:
1) Закомментил в customerPage (th:id="${favouritesGroupList[0].id}"), тк favouritesGroup создается по умолчанию только для основного кастомера
2) В UserServiceImpl в методе activateUser сделал создание кастомера вместо юзера

Kirill-Lebedev committed:
Закоментил обратно случайно раскоменченные названия

Kirill-Lebedev committed:
1)Убрал закоменченную строку в customerPage.html
2)Немного поменял логику вывода ошибок по неправильно введенному email и паролям в MainPageRestController, mainPage.js и registration.js
3)Добавил регистрацию со страницы newsDetails.html

tediik commented on src/main/resources/static/js/mainPage.js:
```diff
@@ -66,14 +66,22 @@ function register() {
+                $("#passwordError").hide();
+            }
+        }
+        /*error: function (data) {
```
лишний код убирай, потом все равно запутаемся зачем он, проще новый написать

tediik commented on src/main/resources/static/js/registration.js:
```diff
@@ -0,0 +1,55 @@
+                $("#passwordError").hide();
+            }
+        }
+        /*error: function (data) {
```
этот код убирай

tediik commented on src/main/resources/static/js/registration.js:
```diff
@@ -0,0 +1,55 @@
+            if (data === "success") {
+                $("#emailConfirmationSent").show();
+                toastr.info('Ссылка для подтверждения регистрации отправлена на вашу почту', {timeOut: 5000});
+                close();
+                document.location.href = "redirect:/";
+            }
+            if (data === "duplicatedEmailError") {
+                $("#duplicatedEmailError").show();
+            }
+            if (data === "notValidEmailError") {
+                $("#notValidEmailError").show();
+            }
+            if (data === "passwordError") {
+                $("#passwordError").show();
+                $("#passwordValidError").hide();
+            }
+            if (data === "passwordValidError") {
+                $("#passwordValidError").show();
+                $("#passwordError").hide();
+            }
+        }
```
этот код 1 в 1 код из файла mainPage.js - их никак нельзя объединить?

tediik commented on src/main/resources/templates/newsDetails.html:
```diff
@@ -20,6 +20,75 @@
 </head>
 <body class="d-flex flex-column min-vh-100">
 <header th:insert="/blocks/header :: header"></header>
+
+<div id="openNewRegistrationModal" class="modal">
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h3 class="modal-title">Регистрация</h3>
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                    <span aria-hidden="true">x</span>
+                </button>
+            </div>
```
смотри, этот код у тебя повторяется во всех файлах ниже, он же одинаковый? Вынеси его во фрагмент и подключай везде этот одинаковый кусок, во всех файлах.

tediik commented on src/main/resources/templates/productPage.html:
```diff
@@ -275,17 +352,19 @@ <h3 style="color: blue">Авторизуйтесь или зарегистрир
 <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
 <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.0/umd/popper.min.js"></script>
-<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.1.0/js/bootstrap.min.js"></script>
+<!--<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.1.0/js/bootstrap.min.js"></script>-->
```
лишний код удаляй

tediik commented on src/main/resources/templates/stocksPage.html:
```diff
@@ -16,6 +16,87 @@
+        </div>
+    </div>
+</div>
+<!--<div th:fragment="header">
+    <div class="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 bg-white border-bottom box-shadow">
+        <h5 class="my-0 mr-md-auto font-weight-normal">Company name</h5>
+        <nav class="my-2 my-md-0 mr-md-3">
+            <a class="p-2 text-dark" href="/">Главная</a>
+            <a class="p-2 text-dark" href="/news">Новости</a>
+        </nav>
+        <a th:if="${#authorization.expression('isAuthenticated()')}" class="btn btn-outline-primary p-1" href="/logout">Выйти</a>
+        <a th:unless="${#authorization.expression('isAuthenticated()')}" class="btn btn-outline-primary p-1" href="/login">Войти</a>
+        <a th:unless="${#authorization.expression('isAuthenticated()')}" class="btn btn-outline-primary p-1" href="/registration">Регистрация</a>
+    </div>
+</div>-->
```
лишнего кода быть не должно

Kirill-Lebedev committed:
Merge branch 'dev' into bugfix/registrationFromAllPages

Kirill-Lebedev committed:
1) Удалил логику регистрации из mainPage.js
2) Подключил registration.js в mainPage.html
3) Подтер закомменченный код. Убрал лишний код из MainPageController.
4) Вынес модалку для регистрации в registration.html и подключил на все страницы, с которых возможна регистрация

Kirill-Lebedev committed:
Убрал закоменченный код из productPage.html
